### PR TITLE
support new js sdk with oauth 2.0

### PR DESCRIPTION
**Pull request by ahilles107**
Hi, 

This commit bring support to new cookie "fbsr_" and update SessionPeresistance acording Facebook solution.

it fully work with my prev PR.

### CODE REVIEW

File: Facebook/FacebookSessionPersistence.php [hunk 1]
```diff
@@ -33,9 +46,15 @@ class FacebookSessionPersistence extends \BaseFacebook
      *
      * @return void
      */
+    
     protected function setPersistentData($key, $value)
     {
-        $this->session->set($this->prefix.$key, $value);
+	    if (!in_array($key, self::$kSupportedKeys)) {
```
Comment (lsmith77) [inline Facebook/FacebookSessionPersistence.php]: indent with 4 spaces

File: Facebook/FacebookSessionPersistence.php [hunk 2]
```diff
@@ -77,4 +110,12 @@ class FacebookSessionPersistence extends \BaseFacebook
             $this->session->remove($k);
         }
     }
+
+    protected function constructSessionVariableName($key) {
```
Comment (lsmith77) [inline Facebook/FacebookSessionPersistence.php]: opening bracket for classes/functions must be on a new line

File: Facebook/FacebookSessionPersistence.php [hunk 3]
```diff
@@ -77,4 +110,12 @@ class FacebookSessionPersistence extends \BaseFacebook
...
+    	return $this->prefix.implode('_', array(
+    		'fb',
+    		$this->getAppId(),
+    		$key
```
Comment (henrikbjorn) [inline Facebook/FacebookSessionPersistence.php]: Missing a comma at the end.

File: Facebook/FacebookSessionPersistence.php [hunk 4]
```diff
@@ -16,13 +17,25 @@ class FacebookSessionPersistence extends \BaseFacebook
...
     private $prefix;
 
+   /**
+    * Identical to the parent constructor, except that
+    * we start a PHP session to store the user ID and
+    * access token if during the course of execution
+    * we discover them.
+    *
+    * @param Array $config the application configuration.
+    * @see BaseFacebook::__construct in facebook.php
```
Comment (stof) [inline Facebook/FacebookSessionPersistence.php:22]: This comment is wrong. The constructor is **not** identical. You should not paste the comment from the FB implementation

File: Facebook/FacebookSessionPersistence.php [hunk 5]
```diff
@@ -5,7 +5,8 @@ namespace FOS\FacebookBundle\Facebook;
 use Symfony\Component\HttpFoundation\Session;
 
 /**
- * Implements Symfony2 session persistence for Facebook.
+ * Extends the BaseFacebook class with the intent of using
+ * PHP sessions to store user ids and access tokens.
```
Comment (stof) [inline Facebook/FacebookSessionPersistence.php]: this change is wrong

File: Facebook/FacebookSessionPersistence.php [hunk 6]
```diff
@@ -33,9 +46,15 @@ class FacebookSessionPersistence extends \BaseFacebook
      *
      * @return void
      */
+
```
Comment (stof) [inline Facebook/FacebookSessionPersistence.php]: this line should be removed

File: Facebook/FacebookSessionPersistence.php [hunk 7]
```diff
@@ -16,13 +17,25 @@ class FacebookSessionPersistence extends \BaseFacebook
...
     public function __construct($config, Session $session, $prefix = self::PREFIX)
     {
         $this->session = $session;
         $this->prefix  = $prefix;
+        $this->session->start();
 
         parent::__construct($config);
     }
+    
+    protected static $kSupportedKeys = array('state', 'code', 'access_token', 'user_id');
```
Comment (stof) [inline Facebook/FacebookSessionPersistence.php]: properties should be defined before methods

File: Facebook/FacebookSessionPersistence.php [hunk 8]
```diff
@@ -48,7 +67,16 @@ class FacebookSessionPersistence extends \BaseFacebook
...
     {
-        return $this->session->get($this->prefix.$key, $default);
+        if (!in_array($key, self::$kSupportedKeys)) {
+	      self::errorLog('Unsupported key passed to getPersistentData.');
+	      return $default;
+	    }
+	    
+	    if ($this->session->has($this->constructSessionVariableName($key))) {
+	      return $this->session->get($this->constructSessionVariableName($key));
+	    } else {
```
Comment (stof) [inline Facebook/FacebookSessionPersistence.php]: you should not use `else` here as the `if` always stops the flow by returning

File: Facebook/FacebookSessionPersistence.php [hunk 9]
```diff
@@ -77,4 +111,13 @@ class FacebookSessionPersistence extends \BaseFacebook
             $this->session->remove($k);
         }
     }
+
+    protected function constructSessionVariableName($key) 
+    {
+    	return $this->prefix.implode('_', array(
```
Comment (willdurand) [inline Facebook/FacebookSessionPersistence.php]: Indent with 4 spaces here.

File: Facebook/FacebookSessionPersistence.php [hunk 10]
```diff
@@ -6,7 +6,7 @@ use Symfony\Component\HttpFoundation\Session;
 
 /**
  * Implements Symfony2 session persistence for Facebook.
- *
+ *
```
Comment (willdurand) [inline Facebook/FacebookSessionPersistence.php]: You should remove this trailing space.

File: Facebook/FacebookSessionPersistence.php [hunk 11]
```diff
@@ -15,11 +15,17 @@ class FacebookSessionPersistence extends \BaseFacebook
 
     private $session;
     private $prefix;
+    protected static $kSupportedKeys = array('state', 'code', 'access_token', 'user_id');
 
+   /**
+    * @param Array $config the application configuration.
```
Comment (willdurand) [inline Facebook/FacebookSessionPersistence.php]: I think `array` does not need uppercase on first letter.

File: Facebook/FacebookSessionPersistence.php [hunk 12]
```diff
@@ -35,7 +41,12 @@ class FacebookSessionPersistence extends \BaseFacebook
      */
     protected function setPersistentData($key, $value)
     {
-        $this->session->set($this->prefix.$key, $value);
+        if (!in_array($key, self::$kSupportedKeys)) {
+          self::errorLog('Unsupported key passed to setPersistentData.');
```
Comment (willdurand) [inline Facebook/FacebookSessionPersistence.php]: 4 spaces here too :)

File: Facebook/FacebookSessionPersistence.php [hunk 13]
```diff
@@ -48,7 +59,17 @@ class FacebookSessionPersistence extends \BaseFacebook
      */
     protected function getPersistentData($key, $default = false)
     {
-        return $this->session->get($this->prefix.$key, $default);
+        if (!in_array($key, self::$kSupportedKeys)) {
+          self::errorLog('Unsupported key passed to getPersistentData.');
```
Comment (willdurand) [inline Facebook/FacebookSessionPersistence.php]: indent with 4 spaces here

File: Facebook/FacebookSessionPersistence.php [hunk 14]
```diff
@@ -59,7 +80,12 @@ class FacebookSessionPersistence extends \BaseFacebook
      */
     protected function clearPersistentData($key)
     {
-        $this->session->remove($this->prefix.$key);
+    	if (!in_array($key, self::$kSupportedKeys)) {
```
Comment (willdurand) [inline Facebook/FacebookSessionPersistence.php]: indent with 4 spaces here

File: Facebook/FacebookSessionPersistence.php [hunk 15]
```diff
@@ -48,7 +59,17 @@ class FacebookSessionPersistence extends \BaseFacebook
      */
     protected function getPersistentData($key, $default = false)
     {
-        return $this->session->get($this->prefix.$key, $default);
+        if (!in_array($key, self::$kSupportedKeys)) {
+            self::errorLog('Unsupported key passed to getPersistentData.');
+            return $default;
+        }
+        
+        if ($this->session->has($this->constructSessionVariableName($key))) {
```
Comment (stof) [inline Facebook/FacebookSessionPersistence.php]: you should avoid constructing it twice

File: Facebook/FacebookSessionPersistence.php [hunk 16]
```diff
@@ -59,7 +80,12 @@ class FacebookSessionPersistence extends \BaseFacebook
      */
     protected function clearPersistentData($key)
     {
-        $this->session->remove($this->prefix.$key);
+        if (!in_array($key, self::$kSupportedKeys)) {
+	        self::errorLog('Unsupported key passed to clearPersistentData.');
```
Comment (lsmith77) [inline Facebook/FacebookSessionPersistence.php]: still a space indent issue here.

**Comment by willdurand** [comment]
I suggest you to create a new PR with one clean commit.

**Pull request closed without merging**